From cf8a831bcb53b60a7b5c4b26dda7646ebc7506d8 Mon Sep 17 00:00:00 2001
From: Stephen Hemminger
Date: Wed, 18 Aug 2010 15:56:46 -0700
Subject: [PATCH] Zebra zserv: bogus conditional

This looks like a bug in original code from misunderstanding
of C rules of evaluation.
---
 zebra/zserv.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/zebra/zserv.c b/zebra/zserv.c
index cb5e411..dc3d432 100644
--- a/zebra/zserv.c
+++ b/zebra/zserv.c
@@ -451,7 +451,7 @@ zsend_route_multipath (int cmd, struct zserv *client, struct prefix *p,
     }
 
   /* Metric */
-  if (cmd == ZEBRA_IPV4_ROUTE_ADD || ZEBRA_IPV6_ROUTE_ADD)
+  if (cmd == ZEBRA_IPV4_ROUTE_ADD || cmd == ZEBRA_IPV6_ROUTE_ADD)
     {
       SET_FLAG (zapi_flags, ZAPI_MESSAGE_DISTANCE);
       stream_putc (s, rib->distance);
-- 
1.7.2.3

